core/src/main/java/contacts/core/log/LoggerRegistry.kt

```diff
@@ -10,18 +10,22 @@ class LoggerRegistry(logger: Logger) {
 
     // Prevent consumers from invoking the listener functions by not having the registry implement
     // it directly.
-    private class Listener(private val logger: Logger) : CrudApi.Listener {
+    private class Listener(private val logger: Logger, private val apiExecutionStartTimeMillis: MutableMap<CrudApi, Long> = mutableMapOf()) : CrudApi.Listener {
 
         override fun onPreExecute(api: CrudApi) {
-            logRedactable(api)
+            apiExecutionStartTimeMillis[api] = System.currentTimeMillis()
+            logger.log(api.redactedCopyOrThis(logger.redactMessages).toString())
         }
 
-        override fun onPostExecute(result: CrudApi.Result) {
-            logRedactable(result)
-        }
+        override fun onPostExecute(api: CrudApi, result: CrudApi.Result) {
+            val executionTime = apiExecutionStartTimeMillis.remove(api)
```

Comment:
I'm an idiot for missing this lol. This is supposed to be something like;

```kotlin
val executionTime = System.currentTimeMillis() - apiExecutionStartTimeMillis.remove(api)
```

I'll fix it in main directly =)